src/ar_http_iface_server.erl

```diff
@@ -60,8 +69,37 @@ do_start(Port) ->
 		},
 
 	{ok, _} =
-		cowboy:start_clear(
-			ar_cowboy_listener,
-			[{port, Port}],
-			ProtocolOpts
-		).
+		case Scheme of
+			http ->
+				cowboy:start_clear(
+					ar_cowboy_listener,
+					[{port, Port}],
+					ProtocolOpts
+				);
+			https ->
+				CustomDomains = ar_meta_db:get(custom_domains),
+				SniHosts = derive_sni_hosts(CustomDomains),
+				cowboy:start_tls(ar_cowboy_listener, [
+					{port, Port},
+					{certfile, "priv/tls/cert.pem"},
+					{keyfile, "priv/tls/key.pem"},
+					{sni_hosts, SniHosts}
+				], ProtocolOpts)
+		end.
+
+get_required_proplist_value(Name, Proplist) ->
+	case proplists:get_value(Name, Proplist) of
```

Comment:
This will break when the property value is `undefined`. You can use lists:keyfind/3 instead.

Reply:
I also thought about this, but I don't think it's a problem

Reply:
Otherwise, I think `prop lists:lookup/2` is a better match. I can update the function if you want, but in this case considering `undefined` as a specified value would only crash later on.